sd/StringParser/test/MainTest.java

```diff
@@ -0,0 +1,52 @@
+import com.company.FileAdapter;
+import com.company.Main;
+import com.company.StringFilter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Scanner;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by SLU on 10/29/2018.
+ */
+public class MainTest {
+    @Before
+    public void setUp() throws Exception {
+    }
+
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void mainTest() throws Exception {
+        File file = Paths.get("test\\tests\\cases.tsv").toFile();
+        Scanner scanner = new Scanner(file);
+        int currentLine = 0;
+
+        while(scanner.hasNextLine()) {
```

Comment:
block looks good, i didnt know that we can use scanner in this way too with delimiters

Reply:
Thank you!